model/src/gateway/payload/update_presence.rs

```diff
@@ -30,61 +30,66 @@ impl UpdateStatusError {
 
     /// Consume the error, returning the owned error type and the source error.
     #[must_use = "consuming the error into its parts has no effect if left unused"]
-    pub fn into_parts(self) -> (UpdateStatusErrorType, Option<Box<dyn Error + Send + Sync>>) {
+    pub fn into_parts(
+        self,
+    ) -> (
+        UpdatePresenceErrorType,
+        Option<Box<dyn Error + Send + Sync>>,
+    ) {
         (self.kind, None)
     }
 }
 
-impl Display for UpdateStatusError {
+impl Display for UpdatePresenceError {
     fn fmt(&self, f: &mut Formatter<'_>) -> FmtResult {
         match self.kind {
-            UpdateStatusErrorType::MissingActivity => {
+            UpdatePresenceErrorType::MissingActivity => {
                 f.write_str("at least one activity must be provided")
             }
         }
     }
 }
 
-impl Error for UpdateStatusError {}
+impl Error for UpdatePresenceError {}
 
-/// Type of [`UpdateStatusError`] that occurred.
+/// Type of [`UpdatePresenceError`] that occurred.
 #[derive(Debug)]
 #[non_exhaustive]
-pub enum UpdateStatusErrorType {
+pub enum UpdatePresenceErrorType {
     /// No activities provided.
     MissingActivity,
 }
 
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct UpdateStatus {
-    pub d: UpdateStatusInfo,
+pub struct UpdatePresence {
+    pub d: UpdatePresenceData,
     pub op: OpCode,
 }
 
-impl UpdateStatus {
-    /// Create a new, valid [`UpdateStatus`] payload.
+impl UpdatePresence {
+    /// Create a new, valid [`UpdatePresence`] payload.
     ///
     /// # Errors
     ///
-    /// Returns an error of type [`UpdateStatusErrorType::MissingActivity`] if
+    /// Returns an error of type [`UpdatePresenceErrorType::MissingActivity`] if
     /// an empty set of activites is provided.
     pub fn new(
         activities: impl Into<Vec<Activity>>,
         afk: bool,
         since: impl Into<Option<u64>>,
         status: impl Into<Status>,
-    ) -> Result<Self, UpdateStatusError> {
-        let d = UpdateStatusInfo::new(activities, afk, since, status)?;
+    ) -> Result<Self, UpdatePresenceError> {
+        let d = UpdatePresenceData::new(activities, afk, since, status)?;
 
         Ok(Self {
             d,
-            op: OpCode::StatusUpdate,
+            op: OpCode::PresenceUpdate,
         })
     }
 }
 
 #[derive(Clone, Debug, Deserialize, Eq, Hash, PartialEq, Serialize)]
-pub struct UpdateStatusInfo {
+pub struct UpdatePresenceData {
```

Comment:
I think `UpdatePresencePayload` or something would be better, "data" isn't very meaningful.